protoc-gen-twirp/generator.go

```diff
@@ -1335,6 +1347,32 @@ func (t *twirp) generateServerProtobufMethod(service *descriptor.ServiceDescript
 	t.P()
 }
 
+func (t *twirp) generateServerInterceptorHandler(service *descriptor.ServiceDescriptorProto, method *descriptor.MethodDescriptorProto) {
+	methName := methodNameCamelCased(method)
+	servName := serviceNameCamelCased(service)
+	inputType := t.goTypeName(method.GetInputType())
+	outputType := t.goTypeName(method.GetOutputType())
+	t.P(`    handler = func(ctx `, t.pkgs["context"], `.Context, req *`, inputType, `) (*`, outputType, `, error) {`)
+	t.P(`      resp, err := s.interceptor(`)
+	t.P(`        func(ctx `, t.pkgs["context"], ` .Context, req interface{}) (interface{}, error) {`)
+	t.P(`          typedReq, ok := req.(*`, inputType, `)`)
+	t.P(`          if !ok {`)
+	t.P(`            return nil, `, t.pkgs["twirp"], `.InternalError("could not convert to a *`, inputType, `")`)
+	t.P(`          }`)
+	t.P(`          return s.`, servName, `.`, methName, `(ctx, typedReq)`)
+	t.P(`        },`)
+	t.P(`      )(ctx, req)`)
+	t.P(`      if resp != nil {`)
+	t.P(`        typedResp, ok := resp.(*`, outputType, `)`)
+	t.P(`        if !ok {`)
+	t.P(`          return nil, `, t.pkgs["twirp"], `.InternalError("could not convert to a *`, outputType, `")`)
```

Comment:
add more context to this error: `twirp.InternalError("failed type assertion resp.(*Msg) when calling interceptor handler")`. This error should never happen, but if it does, we better have a long descriptive error to make it as easy to find as possible (same for the type assertion on the request)

Reply:
Done